Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/FeedbackForm.hs

```diff
@@ -0,0 +1,66 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# LANGUAGE DerivingVia #-}
+
+module Domain.Types.FeedbackForm where
+
+import Domain.Types.Ride (Ride)
+import Kernel.Prelude
+import Kernel.Types.Id
+
+data FeedbackFormReq = FeedbackFormReq
+  { rideId :: Id Ride,
+    feedback :: [FeedbackAnswer]
+  }
+  deriving (Generic, Show, FromJSON, ToJSON, ToSchema)
+
+data FeedbackAnswer = FeedbackAnswer
+  { id :: Id FeedbackAnswer,
```

Comment:
What is this `id`? And why is it require?

Reply:
this will contain all the badges selected by customer for particular category(like Driver, Ride, Vehicle that is sent in the feedback form itself) and this id will be sent to the BAP while submitting the feedbackform to uniquely identify that the particlular selected badges are from which particular category that is sent by the feedback form.